### PR TITLE
validate NR

### PR DESCRIPTION
**Pull request by eve-git**
*Issue #:* /bcgov/entity#14439

*Description of changes:*
Validate NR format. If the input NR number is valid, normalize the NR number and re-write the result to the input field
For example, input 1234567, it is a valid NR, the NR number input field shows 'NR 1234567'

The acceptable NR numbers are like:
NR 1234567
1234567
NR1234567
NR    1234567

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by severinbeauvais** [comment]
Eve. I assigned you so that the checks run.

File: src/components/common/YourCompany/CompanyName/CorrectNameRequest.vue
Comment (severinbeauvais) [inline src/components/common/YourCompany/CompanyName/CorrectNameRequest.vue:171]: What happens if `value.match(VALID_FORMAT)` is not an array with 3 or more items?

Comment (eve-git) [reply src/components/common/YourCompany/CompanyName/CorrectNameRequest.vue:171]: for example?

File: tests/unit/CorrectNameRequest.spec.ts
Comment (severinbeauvais) [inline tests/unit/CorrectNameRequest.spec.ts:111]: Thanks for the nice tests.

Can you please add a test(s) for leading and/or trailing spaces and extra spaces between "NR" and the number? Thanks.

File: src/components/common/YourCompany/CompanyName/CorrectNameRequest.vue
Comment (eve-git) [reply src/components/common/YourCompany/CompanyName/CorrectNameRequest.vue:171]: if the NR matches the regex, it should be an array with 3 items.

Comment (severinbeauvais) [reply src/components/common/YourCompany/CompanyName/CorrectNameRequest.vue:171]: I tried but can't find an example. Since this array indexing only happens if the test is True then I guess it's safe.

File: tests/unit/CorrectNameRequest.spec.ts
Comment (eve-git) [reply tests/unit/CorrectNameRequest.spec.ts:111]: If there are leading or trailing spaces, the NR is invalid so far. Should the spaces be ignored and pass the validation?

Comment (severinbeauvais) [reply tests/unit/CorrectNameRequest.spec.ts:111]: Yes, `  ` is invalid.

However, `  NR  1234567  ` should be  valid.

Comment (eve-git) [reply tests/unit/CorrectNameRequest.spec.ts:111]: @severinbeauvais the test cases have added. Please review.

File: tests/unit/CorrectNameRequest.spec.ts
Comment (severinbeauvais) [inline tests/unit/CorrectNameRequest.spec.ts:145]: Typo: `tearing` should be `trailing`.

Also on line 161.

**Review by severinbeauvais — COMMENTED** [review]
A couple of typos but otherwise LGTM.

Please get some other approvals before merging.

**Review by JazzarKarim — APPROVED** [review]
LGTM 👍

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**
